### PR TITLE
Refactor the API for quant method argument for quantize function

### PR DESCRIPTION
**Pull request by jerryzh168**
Summary:
Addressing feedback from https://github.com/pytorch/ao/issues/384 and https://github.com/pytorch/ao/issues/375

* expanded function api names, e.g. int4_weight_only
* removed string API

Test Plan:
regression tests

python test/quantization/test_quant_api.py
python test/integration/test_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: test/integration/test_integration.py [hunk 1]
```diff
@@ -20,9 +20,9 @@
     DynamicallyPerAxisQuantizedLinear,
 )
 from torchao.quantization.quant_api import (
-    int4wo,
-    int8wo,
-    int8da_int8w,
+    int4_weight_only,
+    int8_weight_only,
+    int8_dynamic_act_int8_weight,
```
Comment (msaroufim) [inline test/integration/test_integration.py:25]: can we expand out act into activation as well? Might as well since we hardly have any abbreviations left

**Review by msaroufim — APPROVED** [review]
cool!

**Comment by supriyar** [comment]
can you also include the update to the main README as well in this PR - specifically the snippet in https://github.com/pytorch/ao?tab=readme-ov-file#without-intrusive-code-changes

**Review by vayuda — APPROVED** [review]
yea i think these function names are more clear and definitely better than the string api

**Pull request merged**